command.go

```diff
@@ -562,9 +562,11 @@ func (cmd *Command) Run(ctx context.Context, osArgs []string) (deferErr error) {
 	}
 
 	if cmd.Before != nil && !cmd.Root().shellCompletion {
-		if err := cmd.Before(ctx, cmd); err != nil {
+		if bctx, err := cmd.Before(ctx, cmd); err != nil {
 			deferErr = cmd.handleExitCoder(ctx, err)
 			return deferErr
+		} else if bctx != nil {
+			ctx = bctx
```

Comment:
Is it not a legitimate usage for one's `Before` to return a new context *and* an error?

Reply:
what would be the purpose of that since the app is going to return an error immediately anyway .

Reply:
Lets implement it if and when someone asks for it.